### ISSUE TITLE
ArcGIS Server Site configuration fails when there is a collection in NodeAgentExt.xml -> NodeAgent.Observers

### ISSUE THREAD

**Issue by marcwilkes** [Esri/arcgis-powershell-dsc#384]
https://github.com/Esri/arcgis-powershell-dsc/blob/35c53bca08644584440a19cb8738e7e307ecd2cf/Modules/ArcGIS/ArcGISUtility.psm1#L577

The above line results in an error, if there are multiple Observer-Nodes within the Observers-Element. This occured while installing ArcGIS Server 10.9.1 plus latest patches. For some reason there was another Observer-Node added to NodeAgentExt.xml.

**Configuration:**
`...
"Installer": {
                "Path": "\\\\ArcGIS Enterprise\\ArcGIS_Server_Windows_1091_de_180043.exe",
                "PatchesDir": "\\\\ArcGIS Enterprise\\",
                "PatchInstallOrder": [
			"ArcGIS-1091-S-SP-Patch.msp",
			"ArcGIS-1091-S-Log4j-Patch.msp"
                ],
                "InstallDir": "C:\\bin\\ArcGIS\\Server",
                "InstallDirPython": "C:\\bin\\Python27"
            }
...`

**Error shown in log: (although in german, you can see that the RemoveChild-Function is not prepared for an array**
`15.03.2022 15:06:43: Das Argument "0" mit dem Wert  "System.Object[]" für "RemoveChild" kann nicht in den Typ "System.Xml.XmlNode" konvertiert werden: "Der Wert "System.Object[]" vom Typ "System.Object[]" kann nicht in den Typ "System.Xml.XmlNode" konvertiert werden."`

**XML:**
![image](https://user-images.githubusercontent.com/69294548/158538918-736e8119-bc8f-4392-a311-463bc8b33269.png)

**Workaround:**
First, I installed, (licensed) and configured the AGS without any patches. The site configuration succeeded (and the amazon-node has been removed). Afterwards, I installed the patches.

**Comment by lorenzmeyer77** [comment]
I observed exactly the same issue. 
For workaround: Un-Install the recent Log4J Patch for the ArcGIS Server and then the DSC runs again.

**Comment by cameronkroeker** [comment]
Hello @marcwilkes,

Thank you for reporting this issue. We plan to provide a fix in an updated release of the module, in the mean time here are two workarounds for v3.3.1:

You will run into the following error when doing a fresh install of 10.9.1 (-Mode InstallLicenseConfigure) that includes the patch, and an upgrade (-Mode Upgrade):

```
Error: Cannot convert argument "0", with value: "System.Object[]", for "RemoveChild" to type "System.Xml.XmlNode": "Cannot convert the "System.Object[]" value of type "System.Object[]" to type "System.Xml.XmlNode"."
```
**Workaround Option 1:** First run `-Mode InstallLicenseConfigure` omitting the log4j patches, then once the deployment has completed, add `ArcGIS-1091-S-Log4j-Patch.msp`, and `ArcGIS-1091-PFA-Log4j-Patch.msp` back to the json and run `-Mode Install`.

> Note: If performing an upgrade, first run `-Mode Upgrade` omitting the log4j patch, then once the deployment has completed, add `ArcGIS-1091-S-Log4j-Patch.msp`, and `ArcGIS-1091-PFA-Log4j-Patch.msp` back to the json and run `-Mode Install`.

**Workaround Option 2:** This is a bit more complex, but is the actual fix we plan to include in the next release, and will allow you to include the log4j patches in InstallLicenseConfigure and Upgrades:

Change:
https://github.com/Esri/arcgis-powershell-dsc/blob/35c53bca08644584440a19cb8738e7e307ecd2cf/Modules/ArcGIS/ArcGISUtility.psm1#L547-L557
To:
```powershell
    if(Test-Path $File){
            [xml]$xml = Get-Content $File
            if((Select-Xml -Xml $xml -XPath "//NodeAgent/Observers/Observer[@platform='amazon']").Length -gt 0 -or (Select-Xml -Xml $xml -XPath "//NodeAgent/Plugins/Plugin[@platform='amazon']").Length -gt 0){
                Write-Verbose "Amazon elements exist in $File"
                $Enabled = $true
            }
    }
```
Change:
https://github.com/Esri/arcgis-powershell-dsc/blob/35c53bca08644584440a19cb8738e7e307ecd2cf/Modules/ArcGIS/ArcGISUtility.psm1#L575-L586
    
To:
```powershell
    if((Select-Xml -Xml $xml -XPath "//NodeAgent/Observers/Observer[@platform='amazon']").Length -gt 0){
                $amazonObserverNode = $xml.NodeAgent.Observers.SelectSingleNode("//Observer[@platform='amazon']")
                if($null -ne $amazonObserverNode){
                    Write-Verbose "Amazon Observer exists in $File. Removing it"
                    $amazonObserverNode.ParentNode.RemoveChild($amazonObserverNode) | Out-Null
                    $Changed = $true
                }
    }
    if((Select-Xml -Xml $xml -XPath "//NodeAgent/Plugins/Plugin[@platform='amazon']").Length -gt 0){
                $amazonPluginNode = $xml.NodeAgent.Plugins.SelectSingleNode("//Plugin[@platform='amazon']")
                if($null -ne $amazonPluginNode){
                    Write-Verbose "Amazon plugin exists in $File. Removing it"
                    $amazonPluginNode.ParentNode.RemoveChild($amazonPluginNode) | Out-Null
                    $Changed = $true
                }
   }
```

Thanks,
Cameron K.

**Comment by sodtom** [comment]
This seems to be issue in Azure ARM+DSC deployments too - the text refers Amazon only.

_Status Message: The resource operation completed with terminal provisioning state 'Failed'. (Code: ResourceDeploymentFailure) - VM has reported a failure when processing extension 'DSCConfiguration0'. Error message: "DSC Configuration 'ServerConfiguration' completed with error(s). Following are the first f
ew: Cannot convert argument "0", with value: "System.Object[]", for "RemoveChild" to type "System.Xml.XmlNode": "Cannot convert the "System.Object[]" value of type "System.Object[]"
to type "System.Xml.XmlNode"." PowerShell DSC resource ArcGIS_Server  failed to execute Set-TargetResource functionality with error message: The operation has timed out  The SendConf
igurationApply function did not succeed."
\+     New-AzResourceGroupDeployment -Name ((Get-ChildItem $TemplateFile ...
\+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-AzResourceGroupDeployment], Exception
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDeploymentCmdlet_

**Comment by cameronkroeker** [comment]
This issue as been addressed in the latest release, v3.3.2:

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v3.3.2

Thanks,
Cameron K.
